genai_processors/contrib/genai_langchain_streaming.py

```diff
@@ -0,0 +1,136 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""
+genai_langchain_processor.py
+
+A lightweight LangChain processor that wraps any BaseChatModel to provide:
+  - Conversation memory across turns
+  - Real-time streaming responses
+  - Multimodal input support (text + images)
+  - Flexible Jinja-based prompt templating
+  - Output metadata preservation
+"""
+
+
+import base64
+from collections.abc import AsyncIterable
+from typing import List, Optional, Union
+
+from genai_processors import processor
+from genai_processors.content_api import (
+    ProcessorContent,
+    ProcessorPart,
+    is_text,
+    is_image,
+)
+from langchain_core.messages import HumanMessage, SystemMessage, AIMessage
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.prompts import ChatPromptTemplate
+
+
+class GenAILangChainProcessor(processor.Processor):
```

Comment:
Maybe call it just `LangChainModel` and the file just `langchain_model.py`? That would be consistent with genai_model.py and  ollama_model.py which are already there.